Failed to build.

I haven't find the doc about how to build Uno.Wasm.Bootstrap. But I found a `build.ps1`, so tried to run it. But the result is
```text
PS D:\Workspace\1_Useful\Uno.Wasm.Bootstrap\build> D:\Workspace\1_Useful\Uno.Wasm.Bootstrap\build\build.ps1
Preparing to run build script...
Running build script...

========================================
Version
========================================
Cake.exe : An error occurred when executing task 'Version'.
At D:\Workspace\1_Useful\Uno.Wasm.Bootstrap\build\build.ps1:234 char:1
+ &$CAKE_EXE $cakeArguments
+ ~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (An error occurr...task 'Version'.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
Error: 
One or more errors occurred.
	
GitVersion: Process returned an error (exit code 1).
```

When I tried to build Uno.Wasm.Bootstrap.sln in VS2017, the result is
```text
1>------ Build started: Project: Uno.Wasm.Bootstrap, Configuration: Debug Any CPU ------
1>C:\Users\cloud\.nuget\packages\microsoft.build.tasks.git\1.0.0-beta-62925-02\build\Microsoft.Build.Tasks.Git.targets(36,5): warning : Submodule 'src/Uno.Wasm.Bootstrap/linker' doesn't have any commit, the source code won't be available via source link.
1>LinkerSteps\ExplicitBlacklistStep.cs(24,12,24,17): error CS0234: The type or namespace name 'Cecil' does not exist in the namespace 'Mono' (are you missing an assembly reference?)
1>LinkerSteps\Extensions.cs(20,12,20,17): error CS0234: The type or namespace name 'Cecil' does not exist in the namespace 'Mono' (are you missing an assembly reference?)
1>LinkerSteps\PreserveTypeConverter.cs(32,12,32,17): error CS0234: The type or namespace name 'Cecil' does not exist in the namespace 'Mono' (are you missing an assembly reference?)
1>LinkerSteps\PreserveUsingAttributesStep.cs(20,12,20,17): error CS0234: The type or namespace name 'Cecil' does not exist in the namespace 'Mono' (are you missing an assembly reference?)
1>ShellTask.cs(28,12,28,17): error CS0234: The type or namespace name 'Cecil' does not exist in the namespace 'Mono' (are you missing an assembly reference?)
1>ShellTask.Linker.cs(28,12,28,17): error CS0234: The type or namespace name 'Cecil' does not exist in the namespace 'Mono' (are you missing an assembly reference?)
1>ShellTask.Linker.cs(29,12,29,17): error CS0234: The type or namespace name 'Cecil' does not exist in the namespace 'Mono' (are you missing an assembly reference?)
1>LinkerSteps\ExplicitBlacklistStep.cs(30,39,30,47): error CS0246: The type or namespace name 'BaseStep' could not be found (are you missing a using directive or an assembly reference?)
1>LinkerSteps\ExplicitBlacklistStep.cs(39,27,39,34): error CS0115: 'ExplicitBlacklistStep.Process()': no suitable method found to override
1>LinkerSteps\Extensions.cs(26,36,26,49): error CS0246: The type or namespace name 'TypeReference' could not be found (are you missing a using directive or an assembly reference?)
1>LinkerSteps\Extensions.cs(68,36,68,49): error CS0246: The type or namespace name 'TypeReference' could not be found (are you missing a using directive or an assembly reference?)
1>LinkerSteps\Extensions.cs(65,30,65,43): error CS0246: The type or namespace name 'TypeReference' could not be found (are you missing a using directive or an assembly reference?)
1>LinkerSteps\PreserveTypeConverter.cs(37,40,37,51): error CS0246: The type or namespace name 'ResolveStep' could not be found (are you missing a using directive or an assembly reference?)
1>LinkerSteps\PreserveUsingAttributesStep.cs(26,47,26,58): error CS0246: The type or namespace name 'ResolveStep' could not be found (are you missing a using directive or an assembly reference?)
1>LinkerSteps\PreserveTypeConverter.cs(54,30,54,44): error CS0246: The type or namespace name 'TypeDefinition' could not be found (are you missing a using directive or an assembly reference?)
1>LinkerSteps\PreserveUsingAttributesStep.cs(50,32,50,46): error CS0246: The type or namespace name 'TypeDefinition' could not be found (are you missing a using directive or an assembly reference?)
1>LinkerSteps\PreserveTypeConverter.cs(73,31,73,45): error CS0246: The type or namespace name 'TypeDefinition' could not be found (are you missing a using directive or an assembly reference?)
1>LinkerSteps\PreserveTypeConverter.cs(39,27,39,34): error CS0115: 'PreserveTypeConverters.Process()': no suitable method found to override
1>LinkerSteps\PreserveUsingAttributesStep.cs(70,31,70,45): error CS0246: The type or namespace name 'TypeDefinition' could not be found (are you missing a using directive or an assembly reference?)
1>LinkerSteps\PreserveUsingAttributesStep.cs(90,33,90,49): error CS0246: The type or namespace name 'MethodDefinition' could not be found (are you missing a using directive or an assembly reference?)
1>LinkerSteps\PreserveUsingAttributesStep.cs(95,27,95,41): error CS0246: The type or namespace name 'TypeDefinition' could not be found (are you missing a using directive or an assembly reference?)
1>LinkerSteps\PreserveUsingAttributesStep.cs(35,27,35,34): error CS0115: 'PreserveUsingAttributesStep.Process()': no suitable method found to override
1>ShellTask.cs(268,50,268,66): error CS0246: The type or namespace name 'EmbeddedResource' could not be found (are you missing a using directive or an assembly reference?)
1>ShellTask.cs(281,37,281,55): error CS0246: The type or namespace name 'AssemblyDefinition' could not be found (are you missing a using directive or an assembly reference?)
1>ShellTask.cs(281,64,281,80): error CS0246: The type or namespace name 'EmbeddedResource' could not be found (are you missing a using directive or an assembly reference?)
1>ShellTask.cs(298,11,298,27): error CS0246: The type or namespace name 'MethodDefinition' could not be found (are you missing a using directive or an assembly reference?)
1>ShellTask.Linker.cs(148,3,148,11): error CS0246: The type or namespace name 'Pipeline' could not be found (are you missing a using directive or an assembly reference?)
1>ShellTask.Linker.cs(165,24,165,31): error CS0246: The type or namespace name 'ILogger' could not be found (are you missing a using directive or an assembly reference?)
1>ShellTask.Linker.cs(171,27,171,44): error CS0246: The type or namespace name 'MessageImportance' could not be found (are you missing a using directive or an assembly reference?)
1>Done building project "Uno.Wasm.Bootstrap.csproj" -- FAILED.
```

--- a/.gitmodules
+++ b/.gitmodules
@@ -1,3 +1,0 @@
-[submodule "src/Uno.Wasm.Bootstrap/linker"]
-	path = src/Uno.Wasm.Bootstrap/linker
-	url = https://github.com/mono/linker.git

--- a/Readme.md
+++ b/Readme.md
@@ -163,4 +163,14 @@ Use the [Templates/Index.html](src/Uno.Wasm.Bootstrap/Templates/Index.html) file
 ### Configuration of the runtime
 - The msbuild property `MonoRuntimeDebuggerEnabled` can be set to `true` to allow for mono to output additional debugging details, and have the debugger enabled (not supported yet by the mono tooling).
 - The msbuild property `RuntimeConfiguration` allows for the selection of the debug runtime, but is mainly used for debugging the runtime itself. The value can either be `release` or `debug`.
-- The msbuild property `MonoWasmSDKUri` allows the override of the default SDK path.
+
+### Updating the mono-sdk build
+The msbuild properties `MonoWasmSDKUri` and `MonoWasmAOTSDKUri` allow the override of the default SDK paths. Paths can be local files.
+
+To select a different sdk build:
+- Navigate to the [Mono-wasm CI](https://jenkins.mono-project.com/job/test-mono-mainline-wasm/)
+- Select a build
+- Click on the "default" configuration
+- On the left click *Azure Artifacts*
+- Copy the `mono-wasm-xxxx.zip` uri to the `MonoWasmSDKUri` property
+- Copy the `wasm-release-Linux-xxx.zip` uri to the `MonoWasmSDKUri` property

--- a/src/Uno.Wasm.Bootstrap/Constants.cs
+++ b/src/Uno.Wasm.Bootstrap/Constants.cs
@@ -6,7 +6,7 @@ namespace Uno.Wasm.Bootstrap
 {
 	internal class Constants
 	{
-		public const string DefaultSdkUrl = "https://jenkins.mono-project.com/job/test-mono-mainline-wasm/1486/label=ubuntu-1804-amd64/Azure/processDownloadRequest/1486/ubuntu-1804-amd64/sdks/wasm/mono-wasm-d2028ba041c.zip";
-		public const string DefaultAotSDKUrl = "https://jenkins.mono-project.com/job/test-mono-mainline-wasm/1486/label=ubuntu-1804-amd64/Azure/processDownloadRequest/1486/ubuntu-1804-amd64/wasm-release-Linux-d2028ba041c3e91bd950bb28ad523158cb8b68b2.zip";
+		public const string DefaultSdkUrl = "https://jenkins.mono-project.com/job/test-mono-mainline-wasm/1534/label=ubuntu-1804-amd64/Azure/processDownloadRequest/1534/ubuntu-1804-amd64/sdks/wasm/mono-wasm-c93db5b0471.zip";
+		public const string DefaultAotSDKUrl = "https://jenkins.mono-project.com/job/test-mono-mainline-wasm/1534/label=ubuntu-1804-amd64/Azure/processDownloadRequest/1534/ubuntu-1804-amd64/wasm-release-Linux-c93db5b04710cfd505c9672ed4f7263319e77abf.zip";
 	}
 }

--- a/src/Uno.Wasm.Bootstrap/ShellTask.cs
+++ b/src/Uno.Wasm.Bootstrap/ShellTask.cs
@@ -53,7 +53,6 @@ namespace Uno.Wasm.Bootstrap
 		[Microsoft.Build.Framework.Required]
 		public string MonoWasmSDKPath { get; set; }
 
-		[Microsoft.Build.Framework.Required]
 		public string PackagerBinPath { get; set; }
 
 		public Microsoft.Build.Framework.ITaskItem[] ReferencePath { get; set; }
@@ -166,9 +165,10 @@ namespace Uno.Wasm.Bootstrap
 
 			var referencePathsParameter = string.Join(" ", _referencedAssemblies.Select(Path.GetDirectoryName).Distinct().Select(r => $"\"--search-path={r}\""));
 
-			int r1 = RunProcess(PackagerBinPath, $"{referencePathsParameter} {Path.GetFullPath(Assembly)}", _distPath);
+			string packagerBinPath = string.IsNullOrWhiteSpace(PackagerBinPath) ? Path.Combine(MonoWasmSDKPath, "packager.exe") : PackagerBinPath;
+			int packagerResults = RunProcess(packagerBinPath, $"{referencePathsParameter} {Path.GetFullPath(Assembly)}", _distPath);
 
-			if(r1 != 0)
+			if(packagerResults != 0)
 			{
 				throw new Exception("Failed to generate wasm layout");
 			}
@@ -184,7 +184,7 @@ namespace Uno.Wasm.Bootstrap
 				var debugOption = this.RuntimeDebuggerEnabled ? "--debug" : "";
 				var aotOption = this.MonoAOT ? $"--aot --emscripten-sdkdir=\"{emsdkPath}\" --builddir=\"{workAotPath}\"" : "";
 
-				int r2 = RunProcess(PackagerBinPath, $"{debugOption} {aotOption} {referencePathsParameter} {Path.GetFullPath(Assembly)}", _distPath);
+				int r2 = RunProcess(packagerBinPath, $"{debugOption} {aotOption} {referencePathsParameter} {Path.GetFullPath(Assembly)}", _distPath);
 
 				if(r2 != 0)
 				{

--- a/src/Uno.Wasm.Bootstrap/UnoInstallSDKTask.cs
+++ b/src/Uno.Wasm.Bootstrap/UnoInstallSDKTask.cs
@@ -86,27 +86,26 @@ namespace Uno.Wasm.Bootstrap
 
 					ZipFile.ExtractToDirectory(zipPath, SdkPath);
 					Log.LogMessage($"Extracted {sdkName} to {SdkPath}");
-
-					if (MonoAOT)
-					{
-						var aotZipPath = SdkPath + ".aot.zip";
-						Log.LogMessage(Microsoft.Build.Framework.MessageImportance.High, $"Downloading {aotUri} to {aotZipPath}");
-						aotZipPath = RetreiveSDKFile(sdkName, aotUri, aotZipPath);
-
-						foreach (var entry in ZipFile.OpenRead(aotZipPath).Entries)
-						{
-							entry.ExtractRelativeToDirectory(SdkPath, true);
-						}
-
-						Log.LogMessage($"Extracted AOT {sdkName} to {SdkPath}");
-
-						if (IsOSUnixLike)
-						{
-							Process.Start("chmod", $"-R +x {SdkPath}");
-						}
-					}
 				}
 
+				if (MonoAOT && !Directory.Exists(Path.Combine(SdkPath, "wasm-cross-release")))
+				{
+					var aotZipPath = SdkPath + ".aot.zip";
+					Log.LogMessage(Microsoft.Build.Framework.MessageImportance.High, $"Downloading {aotUri} to {aotZipPath}");
+					aotZipPath = RetreiveSDKFile(sdkName, aotUri, aotZipPath);
+
+					foreach (var entry in ZipFile.OpenRead(aotZipPath).Entries)
+					{
+						entry.ExtractRelativeToDirectory(SdkPath, true);
+					}
+
+					Log.LogMessage($"Extracted AOT {sdkName} to {SdkPath}");
+
+					if (IsOSUnixLike)
+					{
+						Process.Start("chmod", $"-R +x {SdkPath}");
+					}
+				}
 				//
 				// Disable the packager override as the local updates have been merged into mono master.
 				//

--- a/src/Uno.Wasm.Bootstrap/UnoInstallSDKTask.cs
+++ b/src/Uno.Wasm.Bootstrap/UnoInstallSDKTask.cs
@@ -82,7 +82,7 @@ namespace Uno.Wasm.Bootstrap
 					var zipPath = SdkPath + ".zip";
 					Log.LogMessage($"Using mono-wasm SDK {sdkUri}");
 
-					RetreiveSDKFile(sdkName, sdkUri, zipPath);
+					zipPath = RetreiveSDKFile(sdkName, sdkUri, zipPath);
 
 					ZipFile.ExtractToDirectory(zipPath, SdkPath);
 					Log.LogMessage($"Extracted {sdkName} to {SdkPath}");
@@ -91,7 +91,7 @@ namespace Uno.Wasm.Bootstrap
 					{
 						var aotZipPath = SdkPath + ".aot.zip";
 						Log.LogMessage(Microsoft.Build.Framework.MessageImportance.High, $"Downloading {aotUri} to {aotZipPath}");
-						RetreiveSDKFile(sdkName, aotUri, aotZipPath);
+						aotZipPath = RetreiveSDKFile(sdkName, aotUri, aotZipPath);
 
 						foreach (var entry in ZipFile.OpenRead(aotZipPath).Entries)
 						{
@@ -107,45 +107,49 @@ namespace Uno.Wasm.Bootstrap
 					}
 				}
 
-				// Download the corresponding packager
-				var packagerPath = Path.Combine(SdkPath, "packager_build");
-				var packagerFilePath = Path.Combine(packagerPath, "packager.cs");
+				//
+				// Disable the packager override as the local updates have been merged into mono master.
+				//
 
-				Directory.CreateDirectory(packagerPath);
+				//// Download the corresponding packager
+				//var packagerPath = Path.Combine(SdkPath, "packager_build");
+				//var packagerFilePath = Path.Combine(packagerPath, "packager.cs");
 
-				if (!File.Exists(packagerFilePath))
-				{
-					File.Copy(PackagerOverrideFile, packagerFilePath, true);
-				}
+				//Directory.CreateDirectory(packagerPath);
 
-				PackagerBinPath = Path.Combine(SdkPath, "packager2.exe");
+				//if (!File.Exists(packagerFilePath))
+				//{
+				//	File.Copy(PackagerOverrideFile, packagerFilePath, true);
+				//}
 
-				var projectFile = $@"
-					<Project Sdk=""Microsoft.NET.Sdk"">
-					  <PropertyGroup>
-						<TargetFramework>net462</TargetFramework>
-						<OutputType>Exe</OutputType>
-						<OutputPath>..</OutputPath>
-						<AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
-					  </PropertyGroup>
-						<ItemGroup>
-							<Reference Include=""Mono.Cecil"">
-								<HintPath>{SdkPath}/Mono.Cecil.dll</HintPath>
-							</Reference>
-							<Reference Include=""Mono.Options"">
-								<HintPath>{SdkPath}/Mono.Options.dll</HintPath>
-							</Reference>
-						</ItemGroup>
-					</Project>
-				";
+				//PackagerBinPath = Path.Combine(SdkPath, "packager2.exe");
 
-				PackagerProjectFile = Path.Combine(packagerPath, "packager2.csproj");
-				File.WriteAllText(PackagerProjectFile, projectFile);
+				//var projectFile = $@"
+				//	<Project Sdk=""Microsoft.NET.Sdk"">
+				//	  <PropertyGroup>
+				//		<TargetFramework>net462</TargetFramework>
+				//		<OutputType>Exe</OutputType>
+				//		<OutputPath>..</OutputPath>
+				//		<AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
+				//	  </PropertyGroup>
+				//		<ItemGroup>
+				//			<Reference Include=""Mono.Cecil"">
+				//				<HintPath>{SdkPath}/Mono.Cecil.dll</HintPath>
+				//			</Reference>
+				//			<Reference Include=""Mono.Options"">
+				//				<HintPath>{SdkPath}/Mono.Options.dll</HintPath>
+				//			</Reference>
+				//		</ItemGroup>
+				//	</Project>
+				//";
 
-				var thisPath = Path.Combine(Path.GetDirectoryName(new Uri(GetType().Assembly.Location).LocalPath));
+				//PackagerProjectFile = Path.Combine(packagerPath, "packager2.csproj");
+				//File.WriteAllText(PackagerProjectFile, projectFile);
 
-				File.Copy(Path.Combine(thisPath, "Mono.Cecil.dll"), Path.Combine(SdkPath, "Mono.Cecil.dll"), true);
-				File.Copy(Path.Combine(thisPath, "Mono.Options.dll"), Path.Combine(SdkPath, "Mono.Options.dll"), true);
+				//var thisPath = Path.Combine(Path.GetDirectoryName(new Uri(GetType().Assembly.Location).LocalPath));
+
+				//File.Copy(Path.Combine(thisPath, "Mono.Cecil.dll"), Path.Combine(SdkPath, "Mono.Cecil.dll"), true);
+				//File.Copy(Path.Combine(thisPath, "Mono.Options.dll"), Path.Combine(SdkPath, "Mono.Options.dll"), true);
 			}
 			catch (Exception e)
 			{
@@ -153,15 +157,26 @@ namespace Uno.Wasm.Bootstrap
 			}
 		}
 
-		private void RetreiveSDKFile(string sdkName, string sdkUri, string zipPath)
+		private string RetreiveSDKFile(string sdkName, string sdkUri, string zipPath)
 		{
-			var client = new WebClient();
-			var wp = WebRequest.DefaultWebProxy;
-			wp.Credentials = CredentialCache.DefaultCredentials;
-			client.Proxy = wp;
+			var uri = new Uri(sdkUri);
 
-			Log.LogMessage(Microsoft.Build.Framework.MessageImportance.High, $"Downloading {sdkName} to {zipPath}");
-			client.DownloadFile(sdkUri, zipPath);
+			if (!uri.IsFile)
+			{
+				var client = new WebClient();
+				var wp = WebRequest.DefaultWebProxy;
+				wp.Credentials = CredentialCache.DefaultCredentials;
+				client.Proxy = wp;
+
+				Log.LogMessage(Microsoft.Build.Framework.MessageImportance.High, $"Downloading {sdkName} to {zipPath}");
+				client.DownloadFile(sdkUri, zipPath);
+
+				return zipPath;
+			}
+			else
+			{
+				return uri.LocalPath;
+			}
 		}
 
 		private string GetMonoTempPath()

--- a/src/Uno.Wasm.Bootstrap/build/Uno.Wasm.Bootstrap.targets
+++ b/src/Uno.Wasm.Bootstrap/build/Uno.Wasm.Bootstrap.targets
@@ -66,7 +66,7 @@
 			<Output TaskParameter="PackagerProjectFile" PropertyName="_UnoMonoPackagerProjectFile" />
 		</UnoInstallSDKTask_v0>
 
-		<MSBuild Projects="$(_UnoMonoPackagerProjectFile)" Targets="Restore;Build" />
+		<MSBuild Condition="'$(_UnoMonoPackagerProjectFile)'!=''" Projects="$(_UnoMonoPackagerProjectFile)" Targets="Restore;Build" />
 
 		<ShellTask_v0
 			Assembly="$(IntermediateOutputPath)$(TargetFileName)"

